go/database/mpt/io/archive.go

```diff
@@ -70,6 +69,8 @@ func ExportArchive(directory string, out io.Writer) error {
 		return err
 	}
 
+	ctx := catchInterrupt()
```

Comment:
The function `ExportArchive` is not the top level function and thus, I think, should not handle interrupts. 

Since support for the interrupt is part of the user interface, this should be done by the `main` package, like the [export](https://github.com/Fantom-foundation/Carmen/blob/main/go/database/mpt/tool/export.go) tool or the [sonictool](https://github.com/Fantom-foundation/Sonic/blob/main/cmd/sonictool/main.go). This also enables the tool to coordinate multiple shutdown activities if needed. 

To fix this, a context should be already passed to this function. Wdyt?

Reply:
First I implemented it into `database/mpt/tool/export.go` but we agreed with @kjezek that we will move it to `io` package mainly because testing this feature inside `main` was not easy.
I think your arguments are very much valid, although I think we should keep it in `io` as it is

Reply:
I think there is a way to fulfil both requirements.  You can propagate the context from the top level function, where it also registers for the kill signal.  You can still check for the context being done in each export method in isolation.  The tests can stay where they are now in their correct packages. 

In short, you propagate the context from the top level, everything else stays as it is.

Reply:
Moving `catchInterrupt` would require to move the test to that package aswell to avoid circular dependency. That requires recreating the helper functions for database creation. Do we really want that...?

Reply:
what if the whole `interrupt.go` file is made more generic feature? It is now in the `mpt` directory, but it looks like more generic feature. What if we store it in `common/interrupt/interrupt.go` and make its methods public.  It would solve the circular dependency and I think we may need this functionality in other parts of carmen, all imports, exports, verification, possibly other schemes